src/frontend/src/app/[locale]/(auth)/dashboard/layout.tsx

```diff
@@ -15,8 +15,16 @@ export async function generateMetadata(props: { params: { locale: string } }) {
   };
 }
 
-export default function DashboardLayout(props: { children: React.ReactNode }) {
-  const t = useTranslations('DashboardLayout');
+export default async function DashboardLayout(props: { children: React.ReactNode }) {
+  const t = await getTranslations('DashboardLayout');
+
+  // eslint-disable-next-line no-console
+  console.log('🔹 Fetching user in DashboardLayout...');
+
+  const customerGuid = await getCustomerGuid(); // Now using the cache-based function
```

Comment:
If we get an exception here, we should load an error page in the props.children

Add a new error page within src/components/ui/error_page.tsx and then load it in props.children variable